plugin/pkg/scheduler/api/validation/validation_test.go

```diff
@@ -17,70 +17,60 @@ limitations under the License.
 package validation
 
 import (
+	"errors"
+	"fmt"
 	"testing"
 
 	"k8s.io/kubernetes/plugin/pkg/scheduler/api"
 )
 
-func TestValidatePriorityWithNoWeight(t *testing.T) {
-	policy := api.Policy{Priorities: []api.PriorityPolicy{{Name: "NoWeightPriority"}}}
-	if ValidatePolicy(policy) == nil {
-		t.Errorf("Expected error about priority weight not being positive")
-	}
-}
-
-func TestValidatePriorityWithZeroWeight(t *testing.T) {
-	policy := api.Policy{Priorities: []api.PriorityPolicy{{Name: "NoWeightPriority", Weight: 0}}}
-	if ValidatePolicy(policy) == nil {
-		t.Errorf("Expected error about priority weight not being positive")
-	}
-}
-
-func TestValidatePriorityWithNonZeroWeight(t *testing.T) {
-	policy := api.Policy{Priorities: []api.PriorityPolicy{{Name: "WeightPriority", Weight: 2}}}
-	errs := ValidatePolicy(policy)
-	if errs != nil {
-		t.Errorf("Unexpected errors %v", errs)
-	}
-}
-
-func TestValidatePriorityWithNegativeWeight(t *testing.T) {
-	policy := api.Policy{Priorities: []api.PriorityPolicy{{Name: "WeightPriority", Weight: -2}}}
-	if ValidatePolicy(policy) == nil {
-		t.Errorf("Expected error about priority weight not being positive")
-	}
-}
-
-func TestValidatePriorityWithOverFlowWeight(t *testing.T) {
-	policy := api.Policy{Priorities: []api.PriorityPolicy{{Name: "WeightPriority", Weight: api.MaxWeight}}}
-	if ValidatePolicy(policy) == nil {
-		t.Errorf("Expected error about priority weight not being overflown.")
-	}
-}
-
-func TestValidateExtenderWithNonNegativeWeight(t *testing.T) {
-	extenderPolicy := api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", FilterVerb: "filter", Weight: 2}}}
-	errs := ValidatePolicy(extenderPolicy)
-	if errs != nil {
-		t.Errorf("Unexpected errors %v", errs)
-	}
-}
-
-func TestValidateExtenderWithNegativeWeight(t *testing.T) {
-	extenderPolicy := api.Policy{ExtenderConfigs: []api.ExtenderConfig{{URLPrefix: "http://127.0.0.1:8081/extender", FilterVerb: "filter", Weight: -2}}}
-	if ValidatePolicy(extenderPolicy) == nil {
-		t.Errorf("Expected error about priority weight for extender not being positive")
-	}
-}
-
-func TestValidateMultipleExtendersWithBind(t *testing.T) {
-	extenderPolicy := api.Policy{
-		ExtenderConfigs: []api.ExtenderConfig{
-			{URLPrefix: "http://127.0.0.1:8081/extender", BindVerb: "bind"},
-			{URLPrefix: "http://127.0.0.1:8082/extender", BindVerb: "bind"},
+func TestValidatePolicy(t *testing.T) {
+	tests := []struct {
+		policy   api.Policy
+		expected error
+	}{
+		{
+			policy:   api.Policy{Priorities: []api.PriorityPolicy{{Name: "NoWeightPriority"}}},
+			expected: errors.New("Priority NoWeightPriority should have a positive weight applied to it or it has overflown"),
```

Comment:
Those error message are actually generated by ValidatePolicy. It makes no sense to make them a const var, since those won't be reused elsewhere.